dask/base.py

```diff
@@ -139,11 +141,20 @@ def normalize_function(func):
 
 
 normalize_token = Dispatch()
-normalize_token.register((int, float, str, tuple, list), lambda a: a)
-normalize_token.register(object,
-        lambda a: normalize_function(a) if callable(a) else a)
+normalize_token.register((int, float, str, unicode, bytes, tuple, list,
```

Comment:
One argument is that you shouldn't.  You should call `tokenize(*mytuple)`.  This gets a bit tricky for systems like `dask.imperative` where the user might give you anything.

Perhaps it is better to descend down into all containers.  This will probably kill performance in some cases, but perhaps one should opt-in to performance and away from safety.